docs/how-to-debug-notifications.md

```diff
@@ -0,0 +1,77 @@
+# How to debug notifications
+
+## Get Firebase project id
+
+Two Firebase projects have been created, one for development environnement (_cozy-flagship-app-push-dev_) and one for production environment (_cozy-flagship-app-push-prod_).
```

Comment:
Maybe we should be clearer here.

It happens a lot to develop on a local machine but being connected on a production's cozy. In that case, if we need to test notification, then we should use the push-prod token. 

Also, we can change the "stack environnement" when we log on a "production build" of the flagship app. We're able to chose "other domain" (or enter a mail address linked to a stg or int or dev cozy) but in that case, notification will not work since we'll use the "production token" but for a "non production cozy". 

I think it's acceptable, but it means that to validate the push notification we need to build a "prod" version of the app, but with the "dev token" in order to connect on "non prod cozy". Once it's tested and validated, then we'll be able to deploy the new config on the "prod cozy" and now we should build a "prod version of the app" with the "prod token". 


I'm wrong?

Reply:
Firebase projects are linked to environments :
- on Android by the location of google-services.json (prod version app get the file from android/app/google-services.json and dev version app get the file from android/app/src/debug/google-services.json). It is how it works automatically.
- on iOS by adding the Firebase configuration file to the appropriate target in Xcode

So it is possible to build a "prod" version of the app with a "dev" Firebase conf but not very practical because you need to update Xcode conf or move conf files. What do you think ? Should we find another way ? Maybe using Firebase prod env in cozy int to be able to test a production app with a production Firebase to our int cozy ?

---

On the other side, if you want to connect to a production cozy from local machine: easy. yarn android --variant=release -> Firebase prod and you can connect to a prod cozy with notifications.

Reply:
> Firebase projects are linked to environments :

Yeah but in Cozy's world, environments can mean "cozy stack environment". It's rarely means "localhost dev / prod". 

> So it is possible to build a "prod" version of the app with a "dev" Firebase conf but not very practical because you need to update Xcode conf or move conf files. What do you think ? Should we find another way ? 

I don't think we have a way to handle that correctly, we should only improve the documentation to explain that point. 

Maybe we can find a way to show an alert within the app if we're connected to a "non prod cozy-stack" with the "prod firebase key" (and vice et versa) Is it possible?

Reply:
> On the other side, if you want to connect to a production cozy from local machine: easy. yarn android --variant=release -> Firebase prod and you can connect to a prod cozy with notifications.

Yes, but we lose logs and debug stuff, no?

Reply:
For android, I created 2 product flavors following [this issue](https://github.com/invertase/react-native-firebase/issues/3504#issuecomment-616796121). You can check the last commit.

At the end, we can do : 
- `yarn android --variant=prodDebug` => Firebase in "prod" and app in "dev"
- `yarn android --variant=prodRelease` =>  Firebase in "prod" and app in "prod"
- `yarn android --variant=devDebug` => Firebase in "dev" and app in "dev"
- `yarn android --variant=devRelease` =>  Firebase in "dev" and app in "prod"

`yarn android`does not work anymore because it takes _debug_ as a variant default value and _debug_ does not exist anymore.

Reply:
@Crash-- @Ldoppea what do you think of this solution ?